code/network/multimsgs.cpp

```diff
@@ -2173,76 +2195,98 @@ void process_netgame_update_packet( ubyte *data, header *hinfo )
 
 */
 // send a request or a reply for mission description, if code == 0, request, if code == 1, reply
+// Cyborg17 - Now recursive for longer descriptions!
 void send_netgame_descript_packet(net_addr *addr, int code)
 {
-	ubyte data[MAX_PACKET_SIZE],val;
-	int desc_len;
+	ubyte data[MAX_PACKET_SIZE];
 	int packet_size = 0;
 
-	// build the header
-	BUILD_HEADER(UPDATE_DESCRIPT);
+	// Get this out of the way, because we don't want to waste our time if the addr is bad.
+	Assert(addr != nullptr);
+	if (addr == nullptr) {
+		return;
+	}
 
-	val = (ubyte)code;
-	ADD_DATA(val);	
+	// if we are the client requesting, just send
+	if ( !code ) {
+		// build the header
+		BUILD_HEADER(UPDATE_DESCRIPT);
+
+		ubyte val = 0;
+		ADD_DATA(val);
 
-	if(code == 1){
-		// add as much of the description as we dare
-		desc_len = (int)strlen(The_mission.mission_desc);
-		if(desc_len > MAX_PACKET_SIZE - 10){
-			desc_len = MAX_PACKET_SIZE - 10;
-			ADD_INT(desc_len);
-			memcpy(data+packet_size, The_mission.mission_desc, desc_len);
-			packet_size += desc_len;
-		} else {
-			ADD_STRING(The_mission.mission_desc);
-		}
-	} 
-	
-	Assert(addr != NULL);
-	if(addr != NULL){
 		psnet_send(addr, data, packet_size);
 	}
+	// otherwise, break up description string if necessary and send to client
+	else {
```

Comment:
Somewhat odd style for `else`

Reply:
I've always done commenting that way. Picked up the habit from the freespace2 code base :)